README.md

```diff
@@ -59,6 +59,16 @@ The main difference is if anything fails to match/replace/.., it will throw a `C
 instead of returning `null` (or false in some cases), so you can now use the return values safely relying on the fact that they can
 only be strings (for replace), ints (for match) or arrays (for grep/split).
 
+Additionally the `Preg` class provides match methods that return `bool` rather than `int`, for stricter type safety
+when the number of pattern matches is not useful:
+
+```php
+use Composer\Pcre\Preg;
+
+if (Preg::isMatch('{fo+}', $string, $matches)) // bool
+if (Preg::isMatchAll('{fo+}', $string, $matches, PREG_OFFSET_CAPTURE)) // bool
```

Comment:
Oh I didn't realize you dropped the throw for PREG_OFFSET_CAPTURE.. I guess I'll add that back in, as passing it changes the type of $matches.

Otherwise yes this PR makes sense.

Reply:
Oh apologies, my bad. I had intended to restrict the offset capture only on the Regex class (by copying the Preg code rather than calling it), but completely forgot to do it (or mention that I had removed it).

Reply:
Shouldn't `matchWithOffset` etc require a $matches param? Also, the name implies that you intend to call the method with an $offset param. Wouldn't `matchWithOffsets` or matchGet|ReturnOffsets be more appropriate?

Reply:
Also, to make Preg drop-in complete we would need isMatchWithOffset and isMatchAllWithOffset

Reply:
Ehh yeah the isMatch variants I don't really care, for the odd time you need that doing a `(bool)` cast at call-site sounds reasonable..

> Shouldn't matchWithOffset etc require a $matches param? 

Yes making $matches non nullable makes sense there, good point.

> Also, the name implies that you intend to call the method with an $offset param. Wouldn't matchWithOffsets or matchGet|ReturnOffsets be more appropriate?

Naming things is hard, I did not claim otherwise ;) matchReturnOffsets is probably best. I guess I'll do a last 1.0.0 overwrite for the day 😅 and then call it done.

Reply:
> I guess I'll do a last 1.0.0 overwrite for the day 😅 and then call it done.

I prefer matchWithOffsets personally, and I've crossed my fingers that isMatchWithOffsets makes the cut :)

Reply:
Ahh sorry you meant just add an s to matchWithOffset, yes that makes total sense.

Reply:
https://github.com/composer/pcre/commit/3d322d715c43a1ac36c7fe215fa59336265500f2

Reply:
Hey thanks!